### ISSUE TITLE
Discovery doesn't work if subnet is not /24

### ISSUE THREAD

**Issue by KyGunsAndRadio** [macdylan/sm2uploader#8]
I normally run printers on a separate subnet configured as 10.10.10.0/25 (255.255.255.128). My client PC is also attached directly to that subnet. But when running sm2uploader, discovery mode would not find my printer. I temporarily changed that subnet to /24 (255.255.255.0) and discovery started working. I'm guessing it's because it's trying to broadcast from 10.10.10.255, which isn't valid on a /25 subnet. Perhaps this is a limitation of UDP discovery?

I was able to use the -host switch and force the IP address, but in that case it didn't identify the printer model. Not sure if that's an issue or not.

**Comment by macdylan** [comment]
Yes, currently it will only broadcast to address 255 under the /24 subnet, which I think is the same as luban. I will test and fix it.

**Comment by macdylan** [comment]
I'v published v2.3, Please let me know if it is resolved.

**Comment by KyGunsAndRadio** [comment]
Wow, thanks for the quick response! I tried 2.3. This time it seems to be broadcasting on 10.0.0.0 and failed to find anything. I also just tried changing my subnet back to /24, but it did the same thing. Is it possible to determine what the current configured subnet is and set the broadcast accordingly? May be more trouble than it's worth, I doubt too many people are running networks like mine.

![image](https://github.com/macdylan/sm2uploader/assets/89168030/d93fbb7b-a151-443c-897b-a7439a7ed1d5)
